devfile.yaml

```diff
@@ -0,0 +1,81 @@
+apiVersion: 1.0.0
+metadata:
+  generateName: che-theia-
+
+components:
+  - mountSources: true
+    command:
+      - tail
+      - '-f'
+      - /dev/null
+    memoryLimit: 512Gi
+    type: dockerimage
+    image: 'quay.io/eclipse/che-theia:next'
+    alias: che-theia-next-dev
+    env:
+      - value: 0.0.0.0
+        name: THEIA_HOST
+      - value: '3130'
+        name: THEIA_PORT
+      - value: '0'
+        name: NODE_TLS_REJECT_UNAUTHORIZED
+
+  - mountSources: true
+    memoryLimit: 3Gi
+    type: dockerimage
+    image: 'quay.io/eclipse/che-theia-dev:next'
+    alias: che-dev
+
+  - id: redhat/vscode-yaml/latest
+    type: chePlugin
+
+  - id: che-incubator/typescript/latest
+    memoryLimit: 2048M
+    type: chePlugin
+
+  - id: github/vscode-pull-request-github/latest
+    type: chePlugin
+
+commands:
+
+  - name: build ... current yarn project
+    actions:
+      - workdir: '${fileDirname}'
+        type: exec
+        command: >
+          echo "build from ${fileDirname}" && killall node; yarn || (yarn
+          lint:fix && yarn format:fix && yarn) && echo -e "\e[32mDone.\e[0m
+          build ... from folder ${fileDirname}"
+        component: che-dev
+
+  - name: test-watch ... current yarn project
+    actions:
+      - workdir: '${fileDirname}'
+        type: exec
+        command: >
+          echo "yarn test:watch from ${fileDirname}" && killall node; yarn
+          test:watch
+        component: che-dev
+
+  - name: build current plugin ... and run with che-theia
+    previewUrl:
+      port: 3130
+    actions:
+      - workdir: '${fileDirname}'
+        type: exec
+        command: |
+          project_folder=$PWD
+          while [ ! -e "$project_folder/package.json" ]; do
+            project_folder=${project_folder%/*}
+            if [ "$project_folder" = "" ]; then
+              echo "Could not detect any 'package.json' file from `${PWD}`. Please open any file from your plugin project.";
+              exit 1;
+            fi
+          done
+          SKIP_TEST=true SKIP_FORMAT=true SKIP_LINT=true yarn --cwd=$project_folder
```

Comment:
in one case (when of build it's doing tests/checks and here it's skipping checks
fine for me b/c I like TDD but I don't know if it's the flow for all devs ?

Reply:
Let's see if someone complains :) if not we will figure out :)